src/odemis/acq/millmng.py

```diff
@@ -0,0 +1,512 @@
+            self._exporter = find_fittest_converter(filename)
+            self._fn_bs, self._fn_ext = dataio.splitext(filename)
+            self._log_dir = os.path.dirname(self._log_path)
+
+        # internal class requirements between functions
+        self._pass_duration = []  # one value per milling setting, duration(s) after which drift acquisition starts
+        self._iterations = []  # one value per milling setting, numbers of scans of .roi
+        stage_md = stage.getMetadata()
+        try:
+            self._ion_beam_angle = stage_md[model.MD_ION_BEAM_TO_SAMPLE_ANGLE]
+        except KeyError:
+            raise ValueError("Ion beam angle not defined in stage metadata")
+
+        # Rough estimate of the stage movement speed, for estimating the extra
+        # duration due to movements (copied from _tiledacq.py)
+        self._move_speed = MOVE_SPEED_DEFAULT
+        if model.hasVA(stage, "speed"):
+            try:
+                self._move_speed = (stage.speed.value["x"] + stage.speed.value["y"]) / 2
+            except Exception as ex:
+                logging.warning("Failed to read the stage speed: %s", ex)
+
+        # estimate milling time from the list of milling settings-> millings
+        time_estimate = 0
+        drift_acq_time = 0
+        stage_time = 0
+
+        for setting_nb, setting in enumerate(self._settings):
+            time_estimate += setting.duration.value
+            if setting.dcRoi.value != UNDEFINED_ROI:
+                self._iterations.append(int(numpy.ceil(setting.duration.value / setting.dcPeriod.value)))
+                self._pass_duration.append(setting.duration.value / self._iterations[setting_nb])
+
+                drift_acq_time += self.estimate_drift_time(setting)
+            else:
+                self._iterations.append(1)
+                self._pass_duration.append(setting.duration.value)
+
+        prev_stage_pos_ref = None  # previous stage position used for reference in estimating time
+
+        for site in self.sites:
+            stage_time += self.estimate_stage_movement_time(site, prev_stage_pos_ref)
+            prev_stage_pos_ref = site.pos.value
+
+        self._milling_time_estimate = time_estimate + drift_acq_time  # time_estimate per feature
+        self._remaining_stage_time = stage_time  # total stage movement time
+
+    def cancel(self, future: "Future") -> bool:
+        """
+        Canceler of acquisition task.
+        :param future: the future that will be executing the task
+        :return: True if it successfully cancelled (stopped) the future
+        """
+        logging.debug("Canceling milling procedure...")
+
+        with future._task_lock:
+            if future._task_state == FINISHED:
+                return False
+            future._task_state = CANCELLED
+            future.running_subf.cancel()
+            logging.debug("Milling procedure cancelled.")  # add line number
```

Comment:
What is this comment about? logging automatically indicates the line number if the logging format requests so.
=> drop this comment?